cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -806,32 +850,42 @@ public void waitLeader() {
    * it.
    *
    * @return true if this node has caught up before timeout, false otherwise
+   * @throws CheckConsistencyException if leaderCommitId bigger than localAppliedId a threshold
+   *     value after timeout
    */
-  private boolean waitUntilCatchUp() {
-    long startTime = System.currentTimeMillis();
-    long waitedTime = 0;
-    long leaderCommitId;
+  protected boolean waitUntilCatchUp(CheckConsistency checkConsistency)
+      throws CheckConsistencyException {
+    long leaderCommitId = Long.MIN_VALUE;
     try {
-      leaderCommitId =
-          ClusterDescriptor.getInstance().getConfig().isUseAsyncServer()
-              ? requestCommitIdAsync()
-              : requestCommitIdSync();
-      if (leaderCommitId == Long.MAX_VALUE) {
-        // Long.MAX_VALUE representing there is a network issue
-        return false;
-      }
+      leaderCommitId = config.isUseAsyncServer() ? requestCommitIdAsync() : requestCommitIdSync();
```

Comment:
Maybe we should discuss whether we need to use the leaderCommitId in mid consistency, because, the applied index is updated in one separate thread, which means that in a very short time of one node, the applied index must be smaller than the commit index. 

In the process of writing, the applied index of the local node is always smaller than the commit index, not to mention the comparison between the local applied index and the commit index of the leader. This leads to the need to wait for a heartbeat time before judging whether the local applied index is catching up with the leader commit index.

Reply:
Thanks houliang
When a leader synchronizes logs to more than half of followers, the current log is set to commitid. Therefore, commitId indicates the maximum number of logs synchronized from the leader to followers. We should use this commitid more appropriately.